suite-native/app/src/hooks/useReportDeviceCompromised.ts

```diff
@@ -0,0 +1,37 @@
+import { useSelector } from 'react-redux';
+import { useEffect, useMemo } from 'react';
+
+import * as Sentry from '@sentry/react-native';
+
+import { selectFirmwareRevisionCheckError } from '@suite-native/settings';
+import { getFirmwareVersion } from '@trezor/device-utils';
+import { selectSelectedDevice } from '@suite-common/wallet-core';
+
+const reportCheckFail = (checkType: 'Firmware revision', contextData: any) => {
+    Sentry.captureException(`${checkType} check failed! ${JSON.stringify(contextData)}`);
+};
+
+const useCommonData = () => {
+    const device = useSelector(selectSelectedDevice);
+    const model = device?.features?.internal_model;
+    const revision = device?.features?.revision;
+    const version = getFirmwareVersion(device);
+    const vendor = device?.features?.fw_vendor;
+
+    return useMemo(
```

Comment:
There is no calculation, so what is the benefit of `useMemo` here?

Reply:
I guess I get it, it is supposed to prevent dependencies change for the useEffect reporting check fail below. Ok then.